sdk/core/core-http/src/policies/systemErrorRetryPolicy.ts

```diff
@@ -175,7 +176,7 @@ function retry(
     return utils
       .delay(retryData.retryInterval)
       .then(() => policy._nextPolicy.sendRequest(request.clone()))
-      .then((res) => retry(policy, request, res, retryData, err))
+      .then((res) => retry(policy, request, res))
```

Comment:
instead of calling retry again here, can't we skip this line? A successful response would then chain back up as the promise resolution value.

Reply:
Yeah.  Removed.  I also converted this function to be async.  It reads better and I hope perf impact is not significant.